### DISCUSSION TITLE
[Feature] Edited pictures should get the same attributes than the original picture

### DISCUSSION THREAD

**Discussion by Frava735** [immich-app/immich#11542] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When using the new picture/edit feature, the resulting picture is saved without the original picture info I.e, time, date, location, album it was part of, ....

Where it should be saved with :
- the same date and time when picture has been taken in the past.
--with the same location (exif info) if location was already stored in the original picture.
- added to the albums the original picture was, because usually editing is mainly used to turn pictures from landscape to portrait and vice/versa, so remaining in the albums theboriginal picturewas.. Cropped if needed.

It is mich easier find the picture and be sure it is part of the same albums, than editing date of the new edited pictured to bring it back in time where it is supposed to be,  and also find out what albums it is part of.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by blue-kaleidoscope** [comment]
I would even go a step further: Why not having the same behaviour as on Adobe Lightroom or iOS (Android is probably similar)? You edit a picture and this asset is now edited. You can always revert all editing operations and go back to the original. When downloading the picture you download it according to the current edit state but internally the original picture is always kept and what the user sees is only a representation of the current edit state.

This would get rid of any duplicates handling (i.e. copying metadata, putting them into the same album they were before etc.).

**Reply by Frava735** [reply]
That would be great. ;-)
But propabaly à lot of work.

And, from my understanding Immich is not an editing tool. Could be achieved by creating an Immich Adobe plug-in... (even if Immich and Lightroom compete on some features).

So Maybe step by step 
- allow the user to duplicate automatically the attributes (create the original state you were mentioning)
- keep track of the attributes edited (allowing to revert back)
- Adobe plug-in for the rest of editing ? Or should I say DarkTable instead in order to maintain the whole open-source ecosystem  ;-)

**Reply by PANICBUTT0N** [reply]
I would really like this feature. I have a lot of images on my iPhone that I have edited, but only the edited versions get uploaded to Immich. So if I trim a video for example, only the trimmed part is uploaded to Immich, and I lose the rest of the video.

**Comment by blue-kaleidoscope** [comment]
Maybe I was misunderstood. I do not expect Immich to replace Lightroom's editing features. I understand that Immich wants to offer basic image editing and the cropping feature is the first step to that which I think is great!

What I personally do not like is that each editting operation creates a new (edited) duplicate. This is where I think iOS (and probably Android) and Lightroom (and probably similar software) have a convenient behaviour. The image object stays the same no matter the editing. 

I don't think a plug-in is necessary.

However, even the feature I am describing is already quite complex.

**Reply by Frava735** [reply]
I agree with you !

I was using Monument 2 from Monument Labs, for storing and managing my pictures and had the behavior I described initially. 

I think, yours is better.

Having the behavior you described (mine being covered by it), solves the issue/feature request.

Let's hope this can be added to the roadmap ;-)

**Comment by mikerjensen** [comment]
Yes, it would be amazing if cropped photos could inherit the relevant EXIF data from the original. Lack of this feature is the only thing that stops me from switching over to Immich for all my image management needs.

**Comment by mikerjensen** [comment]
I have some coder friends who I could perhaps try to persuade to work on just the basic issue (maintaining the EXIF information in edited images). Are there any pointers to what would be involved in getting to grips with this problem? I seem to recall was some mention in another discussion channel that the current architecture of the app does not allow for preserving the EXIF information for the edited image, is that correct?

**Comment by bo0tzz** [comment]
#11470

**Comment by mfrisina6600** [comment]
Chiming in, this would be a fantastic feature to have—I’d go as far as to say it’s a necessary feature to have, if immich aims to act as a true alternative to one of the primary photos apps like iCloud Photos or Google Photos. I hope it comes soon!

**Comment by Vogeluff** [comment]
also just wanna say this is a key feature in my opinion as i wanna replace google photos with immich and rotating photos in my eyes is more about presenting an image, not even editing it. cropping and stuff are also great but just roation is a necessity for me and it shouldnt be complicated and destroy the order of my images or create a duplicate photo at the todays date.
